tests/integration/utils/mock_requests.js

```diff
@@ -24,27 +24,35 @@ function mockAllRequests(fakeServer, mock) {
       if (contents && contents.length) {
         for (const content of contents) {
           if (content && content.init) {
-            // fakeServer.respondWith("GET", content.init.url, (req) => {
-            //   req.respond(200, {
-            //     "Content-Type": content.init.contentType,
-            //   }, content.init.data);
-            // });
-            fakeServer.respondWith("GET", content.init.url,
-              [200, {
-                "Content-Type": content.init.contentType,
-              }, content.init.data]);
+            if (typeof content.init.data === "function") {
+              fakeServer.respondWith("GET", content.init.url, (xhr) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Define this function outside of a loop. ([javascript:FunctionDefinitionInsideLoop](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AFunctionDefinitionInsideLoop))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=308&issues=AWWvpxA-ashydt7kRYLh&open=AWWvpxA-ashydt7kRYLh)

[ISSUE_KEY]: AWWvpxA-ashydt7kRYLh